scripts/addcmakeppa.sh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env sh
+
+ACCEPTED_UBUNTU_CODENAMES="focal
+bionic
+xenial"
+
+printf "%s" "$ACCEPTED_UBUNTU_CODENAMES" | grep -qw "$1"
+if [ $? -ne 0 ]; then
+    echo "'$1' is not a distribution supported by CMake PPA" 1>&2
+    exit 1
+fi
+
+printf "Adding CMake PPA for %s\\n" "$1"
+
+# Exit immediately if one of the commands fail below
+trap exit ERR
+
+# Commands taken from https://apt.kitware.com/
+
+sudo apt-get update -qq
+sudo apt-get install -y apt-transport-https ca-certificates gnupg software-properties-common wget
+
+wget -O - https://apt.kitware.com/keys/kitware-archive-latest.asc 2>/dev/null | gpg --dearmor - | sudo tee /etc/apt/trusted.gpg.d/kitware.gpg >/dev/null
+
+sudo apt-add-repository "deb https://apt.kitware.com/ubuntu/ $1 main"
+sudo apt-get update -qq
```

Comment:
I have noticed that the commands in the Drone CI workflow do not have `sudo` calls.  
Does that also make them unnecessary here?

Reply:
We want to avoid sudo if we can, especially hidden inside a script. Does it work if you remove the "sudo"?

Reply:
The `trusted.gpg.d` directory looks like this on my system, so I'd assume that is also the case in the CI.
```
drwxr-xr-x 2 root root 4096 Oct  6 18:36 trusted.gpg.d
```
That'd mean the `tee` command would definitely fail, right?

Reply:
Typically, in a container, you are root by default.

Reply:
I see, I'll remove the `sudo` calls then. Thanks for the head up!